scripts/flights2.py

```diff
@@ -0,0 +1,727 @@
+                    f"But got:\n{mapping!r}"
+                )
+                raise TypeError(msg)
+        return cls(range, **rest)
+
+    @property
+    def name(self) -> str:
+        """
+        Encodes a short form of ``n_rows`` into the file name.
+
+        Examples
+        --------
+        Note that the final name depends on ``suffix``:
+
+            | n_rows         | stem          |
+            | -------------- | ------------- |
+            | 10_000         | "flights-10k" |
+            | 1_000_000      | "flights-1m"  |
+            | 12_000_000_000 | "flights-12b" |
+        """
+        frac = self.n_rows // 1_000
+        if frac >= 1_000_000:
+            s = f"{frac // 1_000_000}b"
+        elif frac >= 1_000:
+            s = f"{frac // 1_000}m"
+        elif frac >= 1:
+            s = f"{frac}k"
+        else:
+            raise TypeError(self.n_rows)
+        return f"{self._name_prefix}{s}{self.suffix}"
+
+    @property
+    def sort_by(self) -> Column:
+        """Temporal column used to sort the transformed data."""
+        return "time" if "time" in self.columns else "date"
+
+    def transform(self, ldf: pl.LazyFrame, /) -> pl.DataFrame:
+        """
+        Materialize the spec for export.
+
+        Parameters
+        ----------
+        ldf
+            Cleaned source data, spanning ``self.range``.
+        """
+        if dt_format := self.dt_format:
+            ldf = _transform_temporal(ldf, dt_format)
+        ldf = ldf.select(self.columns)
+        if n_rows := self.n_rows:
+            return ldf.collect().sample(n_rows).sort(self.sort_by)
+        return ldf.sort(self.sort_by).collect()
+
+    def write(self, df: pl.DataFrame, output_dir: Path, /) -> None:
+        """
+        Export the materialized spec..
+
+        Parameters
+        ----------
+        df
+            Materialized spec data, the result of ``self.transform(...)``.
+        output_dir
+            Output directory.
+        """
+        fp: Path = output_dir / self.name
+        fp.touch()
+        msg = f"Writing {fp.as_posix()!r} ..."
+        logger.info(msg)
+        match self.suffix:
+            case ".arrow":
+                df.write_ipc(fp, compression="zstd")
+            case ".csv":
+                df.write_csv(
+                    fp,
+                    date_format=None,
+                    datetime_format=None,
+                    time_format=None,
+                    null_value=None,
+                )
+            case ".json":
+                df.write_json(fp)
+            case ".parquet":
+                df.write_parquet(fp, compression="zstd", compression_level=22)
+            case _:
+                fp.unlink()
+                msg = f"Unexpected extension {self.suffix!r}"
+                raise NotImplementedError(msg)
+
+
+class SourceMap:
+    """Handles resource sharing and reading."""
+
+    def __init__(self, input_dir: Path, /) -> None:
```

Comment:
### TODO
- [ ] Add a more meaningful doc

Reply:
Resolved in (https://github.com/vega/vega-datasets/pull/645/commits/859975ebcdb841ee0eabe912ae6081c9431fe955)